traitsui/qt4/ui_panel.py

```diff
@@ -329,12 +329,12 @@ def _fill_panel(panel, content, ui, item_handler=None):
     panel.setCurrentIndex(active)
 
 
-def _size_hint_wrapper(f, ui):
+def _size_hint_wrapper(ui):
     """Wrap an existing sizeHint method with sizes from a UI object.
     """
 
     def sizeHint():
-        size = f()
+        size = QtCore.QSize(-1, -1)
```

Comment:
An alternative would be to do this instead, assuming no one is mutating the width and height after the UI is created:
```
        if ui.view.width > 0 or ui.view.height > 0:
            self.control.sizeHint = _size_hint_wrapper(ui)
```

> A bigger change would be to subclass the various QObjects that are created in this module so that they can take the ui as an argument and override their sizeHint methods.

I am afraid that would be too much.

Reply:
> An alternative would be to do this instead, assuming no one is mutating the width and height after the UI is created:

Darn, but then one may set just one of width and height. What are these for?

Reply:
The view width and height allow the developer to specify preferred width and/or height for the view, overriding the defaults.  For a top level view, this would correspond to the size of the window, for a panel it would be the size of the panel.  The user may only care about restricting the width or the height, depending on the situation.  And these are only "preferred", an embedded UI may get more or less space; and a window may be resized by the user, for example.